azavea/district-builder-dtl-pa
# Update Signup Form with Number of Users on Account

## Overview

This PR adds a new field to the sign up form that allows collecting data for the number of users on the account.

Acceptance criteria:
 * Add a field to the signup form that takes an integer as entry
 * Title: "Number of Users on this Account"
 * Help text: "How many users/students share this account? If it's just you, enter 1."
 * Save this into the database for that user
 * The field should be non-nullable, so set existing users values to 1
 * Add a column to the Django admin users list

### Checklist


### Demo

![Screenshot_2019-08-28 DistrictBuilder Web-based Open Source Software for Collaborative Redistricting(1)](https://user-images.githubusercontent.com/6386/63889251-a3234180-c9ae-11e9-8a00-7a72245f897f.png)

![Screenshot_2019-08-28 Select user to change Redistricting site admin](https://user-images.githubusercontent.com/6386/63889304-bcc48900-c9ae-11e9-83e0-bbd64f9c749f.png)

### Notes

When we eventually run an export for this contest, we want to make sure we also export this field. I wanted to ensure this will work ahead of time, so constructed the query. Placing it here so we can refer to it when the time comes:
```sql
SELECT 
  ps.id, ps.first_name, ps.last_name, ps.email_address, ps.team_member_names, 
  ps.team_name, ps.phone_number, ps.county, ps.zip_code, ps.contest_division, 
  ps.values_statement, ps.plan_id, ps.submitting_user_id, ps.submitted_plan_name, 
  ps.organization_name, ps.values_choices, ps.endorsers, p.num_users_on_account 
FROM redistricting_plansubmission ps 
INNER JOIN redistricting_profile p 
ON ps.submitting_user_id = p.user_id;
```

## Testing Instructions

 * `vagrant up`, `vagrant ssh`
 * `./scripts/update` (to run migrations)
 * `./scripts/server`
 * Browse to `http://localhost:8080/`
 * Sign up for a new user account, and ensure that you must enter a positive integer for the num users field
 * When logged into the app, verify that you can edit your user profile, and the num users field may be viewed and edited (with the same validation logic)
 * Log in as an admin (username/password are in `.env`), browse to the `Users` section, and ensure that a num users field is displayed, and that all old records are showing a 1 as the value.

Closes #156
